### PR TITLE
Where the heck am I??

### PR DESCRIPTION
**Pull request by cbolles**
Add the ability for the BMS to detect the system it is attached to (either charging station or bike). This is done by checking for the heatbeat of either the charger or the pre-charger board. This is added into the existing state machine.

### CODE REVIEW

File: include/BMS/dev/SystemDetect.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+     * @param[in] bikeHeartBeat The heart beat CANopen ID associated with the
+     *                          bike.
+     * @param[in] chargeHeatBeat The heat beat CANopen ID associated with the
+     *                           charger
+     * @param[in] timeout The timeout which represents time between getting
+     *                    heart beat values where the device still recognizes
+     *                    the system it is attached to. If the heart beat is
+     *                    not received within this timeout, device assumes it
+     *                    does not know what the system is attached to. This
+     * heart beat has been detected
```
Comment (mjmagee991) [inline include/BMS/dev/SystemDetect.hpp:47]: NIT: Alignment got messed up here

File: targets/BMS/main.cpp [hunk 2]
```diff
@@ -75,9 +96,18 @@ int main() {
...
     EVT::core::types::FixedQueue<CANOPEN_QUEUE_SIZE, IO::CANMessage> canOpenQueue;
 
+    // Intialize the system detect
+    BMS::DEV::SystemDetect systemDetect(BIKE_HEART_BEAT, CHARGER_HEART_BEAT,
+                                        DETECT_TIMEOUT);
+
+    // Create struct that will hold CAN interrupt parameters
+    struct CANInterruptParams canParams = {
+        .queue = &canOpenQueue,
+        .systemDetect = &systemDetect};
```
Comment (mjmagee991) [inline targets/BMS/main.cpp:106]: NIT: Would look cleaner if there was a comma at the end of this line

**Review by mjmagee991 — CHANGES_REQUESTED** [review]
Nice work. The overall system looks great and is following your interface perfectly. I just had 2 little formatting nitpicks.

**Review by zsteig — APPROVED** [review]

File: targets/BMS/main.cpp (hunk 2 above)
Comment (cbolles) [reply targets/BMS/main.cpp]: Good catch

File: include/BMS/dev/SystemDetect.hpp (hunk 1 above)
Comment (cbolles) [reply include/BMS/dev/SystemDetect.hpp]: Nice find

**Review by mjmagee991 — APPROVED** [review]
Nice

**Pull request merged**